main/src/ca/uwaterloo/flix/language/errors/ResolutionError.scala

```diff
@@ -1089,6 +1089,82 @@ object ResolutionError {
     override def explain(formatter: Formatter): Option[String] = None
   }
 
+  /**
+    * An error raised to indicate a struct does not exist in a `new Struct { ... } @ r` expression
+    *
+    * @param struct the names of the struct
+    * @param loc the location where the error occurred.
+    */
+  case class NonExistentStruct(struct: String, loc: SourceLocation) extends ResolutionError with Recoverable {
+    override def summary: String = s"Nonexistent struct type"
+
+    def message(formatter: Formatter): String = {
+      import formatter._
+      s""">> A struct type with the name `${struct}` does not exist
+         |
+         |${code(loc, "nonexistent struct")}
+         |""".stripMargin
+    }
+  }
+
+  /**
+    * An error raised to indicate a struct is missing a required field in `struct.field` or `struct.field = value` expression
+    *
+    * @param field the names of the missing fields
+    * @param loc the location where the error occurred.
+    */
+  case class NonExistentStructField(field: String, loc: SourceLocation) extends ResolutionError with Recoverable {
+    override def summary: String = s"Struct is missing a field"
```

Comment:
This puts the blame of the struct declaration, but we should blame the incorrect access, right?

I'd make it something like `Struct $name does not have the field $field` (like you have below)

Note we'll need an extra parameter for that

Reply:
Was this fixed?

Reply:
Oops - I had forgotten to fix the summary, I just fixed the message